File: Bai_18/main.py
def tongdoan(l, r, a):
    s = 0
    for i in range(l, r + 1):
        s += a[i]
    return s

with open('E:\Language\Baitaptonghop\Bai_18\THICONG.INP', 'r') as file_input:
    n, k = map(int, file_input.readline().split())
    a = list(map(int, file_input.readline().split()))

maxx = sum(a[:k])  # Sum of the first k elements
cnt = k - 1
s = maxx
d = a[cnt]

for i in range(n - 1, n - k - 1, -1):
    s = s - d + a[i]
    if s > maxx:
        maxx = s
    if cnt > 0:
        cnt -= 1
    d = a [cnt]

with open('E:\Language\Baitaptonghop\Bai_18\THICONG.OUT', 'w') as file_output:
    file_output.write(str(maxx))

File: Bai_18/main.cpp
#include <bits/stdc++.h>
using namespace std;
long long tongdoan(int l, int r, const vector<long long> &a)
{
    long long s = 0;
    for (int i = l; i <= r; i++)
        s = s + a[i];
    return s;
}
long long n, k, maxx, cnt, d, s;
int main()
{
    ios_base::sync_with_stdio(0);
    cin.tie(0);
    cout.tie(0);
    freopen("THICONG.INP", "r", stdin);
    freopen("THICONG.INPOUT", "w", stdout);
    cin >> n >> k;

    vector<long long> a(n);
    for (int i = 0; i < n; i++)
        cin >> a[i];

    maxx = tongdoan(0, k - 1, a);
    cnt = k - 1;
    s = maxx;
    d = a[cnt];

    for (int i = n - 1; i >= n - k; i--)
    {
        s = s - d;
        s = s + a[i];
        if (s > maxx)
            maxx = s;
        if (cnt > 0)
            cnt--;
        d = a[cnt];
    }

    cout << maxx;
    return 0;
}